main.go

```diff
@@ -142,7 +161,21 @@ func main() {
 		RHOBSSecretName:              fmt.Sprintf("%v-prom-remote-write", addonName),
 		RHOBSEndpoint:                envVars[rhobsEndpointEnvVarName],
 		RHSSOTokenEndpoint:           envVars[rhssoTokenEndpointEnvVarName],
-	}).SetupWithManager(mgr, nil); err != nil {
+		NetworkType:                  controllers.SDNNetworkType,
+	}
+
+	err = reconciler.Client.Get(ctx, client.ObjectKeyFromObject(egressNetworkPolicyCRD), egressNetworkPolicyCRD)
```

Comment:
1. Please use the following instead of using `unstructured.Unstructured{}` 
2. Please do not mix error handling with program logic
3. have selection logic in one place instead of trying to override a "default" value - it makes following the code simpler and the reader do not need to make an assumption about the value in the non-overriding case.

```
	crd := apiextensionsv1.CustomResourceDefinition{}
	err := r.Client.Get(ctx, types.NamespacedName{Name: "egressnetworkpolicies.network.openshift.io"}, &crd)
	if  err == nil && !errors.IsNotFound(err){
		setupLog.Error(err, "Error Getting CRD for EgressNetworkPolicy")
		os.Exit(1)
        }
        netInterfaceType := ""
        if crd.UID != "" {
                netInterfaceType := NetworkInterfaceSDN
        } else {
                netInterfaceType := NetworkInterfaceOVN
        }
```
(code adjusted to the naming requests from prev comments)